### DISCUSSION TITLE
Escaping values in vc env pull

### DISCUSSION THREAD

**Discussion by Prioe** [vercel/vercel#4391] [General] [answered]
I've ran into problems trying to add JSON data to Vercel environment.

If you add JSON data to the environment with `vc env add`, the data will be parseable on Vercel - production/preview, but it will be invalid locally after `vc env pull`.

I've created a repo with a minimal reproduction of the issue:
https://github.com/Prioe/now-env-pull-repro

This is the line which causes the issue:
https://github.com/zeit/now/blob/f4d86cb9df757d1154a43549290b6d4a22b55caf/packages/now-cli/src/commands/env/pull.ts#L75

A solution would be to just not escape `"`, since dotenv allows double quotes inside double quotes:
https://github.com/motdotla/dotenv#rules

> inner quotes are maintained (think JSON)

If this is a feasible solution I would be happy to create a pull request with the changes.

**Comment by TooTallNate** [comment]
Thanks for the report. A pull request would be very welcome!

**Comment by styfle** [comment, marked as answer]
Thanks, we'll get this fixed 👍 

I created PR #4412.

**Comment by tonyxiao** [comment]
There is another issue. vercel uses dotenv expand so if one of my values has a `$` in there (e.g. a password) then vercel env pull will result in invalid value. The solution would be to escape `$` as `\$`
